vald/evm/evm.go

```diff
@@ -466,69 +464,50 @@ func getLatestFinalizedBlockNumber(client rpc.Client, confHeight uint64) (*big.I
 	return big.NewInt(int64(blockNumber - confHeight + 1)), nil
 }
 
-func (mgr Mgr) validate(client rpc.Client, txID common.Hash, confHeight uint64, validateTx func(tx *geth.Transaction, txReceipt *geth.Receipt) bool) bool {
-	tx, _, err := client.TransactionByHash(context.Background(), txID)
-	if err != nil {
-		mgr.logger.Debug(sdkerrors.Wrap(err, "get transaction by hash call failed").Error())
-		return false
+func (mgr Mgr) getTxReceiptIfFinalized(chain nexus.ChainName, txID common.Hash, confHeight uint64) (*geth.Receipt, error) {
+	client, ok := mgr.rpcs[strings.ToLower(chain.String())]
+	if !ok {
+		return nil, fmt.Errorf("rpc client not found for chain %s", chain.String())
 	}
 
 	txReceipt, err := client.TransactionReceipt(context.Background(), txID)
-	if err != nil {
-		mgr.logger.Debug(sdkerrors.Wrap(err, "get transaction receipt call failed").Error())
-		return false
+	if err == ethereum.NotFound {
+		return nil, nil
 	}
-
-	if !isTxSuccessful(txReceipt) {
-		mgr.logger.Debug(fmt.Sprintf("transaction %s failed", txReceipt.TxHash.String()))
-		return false
+	if err != nil {
+		return nil, sdkerrors.Wrap(err, "failed getting transaction receipt")
 	}
 
 	latestFinalizedBlockNumber, err := getLatestFinalizedBlockNumber(client, confHeight)
 	if err != nil {
-		mgr.logger.Debug(sdkerrors.Wrap(err, "get latest finalized block number failed").Error())
-		return false
+		return nil, sdkerrors.Wrap(err, "failed getting latest finalized block number")
 	}
 
 	if latestFinalizedBlockNumber.Cmp(txReceipt.BlockNumber) < 0 {
-		mgr.logger.Debug(fmt.Sprintf("transaction %s is not finalized yet", txReceipt.TxHash.String()))
-		return false
+		return nil, nil
```

Comment:
The debug log is still useful here, for e.g if the validator's RPC node is falling out of sync causing this issue